nf_core/pipeline-template/{{cookiecutter.name_noslash}}/lib/JSON.groovy

```diff
@@ -0,0 +1,117 @@
+/*
+ * This file holds several functions used to perform JSON parameter validation, help and summary rendering for the nf-core pipeline template.
+ */
+
+import groovy.json.JsonSlurper
+
+class JSON {
+    /*
+    * This method tries to read a JSON params file
+    */
+    private static LinkedHashMap params_get(String path) {
+        def usage = new LinkedHashMap()
+        try {
+            usage = params_try(path)
+        } catch (Exception e) {
+            println "Could not read parameters settings from JSON. $e"
+            usage = new LinkedHashMap()
+        }
+        return usage
+    }
+
+    /*
+    Method to actually read in JSON file using Groovy.
+    Group (as Key), values are all parameters
+        - Parameter1 as Key, Description as Value
+        - Parameter2 as Key, Description as Value
+        ....
+    Group
+        -
+    */
+    private static LinkedHashMap params_try(String path) throws Exception {
+
+        def json = new File(path).text
+        def Map usage = (Map) new JsonSlurper().parseText(json).get('properties')
+
+        /* Tree looks like this in nf-core schema
+        *  properties <- this is what the first get('properties') gets us
+             group 1
+               properties
+               description
+             group 2
+               properties
+               description
+             group 3
+               properties
+               description
+        */
+        def output_map = new LinkedHashMap()
+
+        //Lets go deeper
+        usage.each { key, val ->
+            //println "Group: $key\n" //Gets the Group names
+            def Map submap = usage."$key".properties //Gets the property object of the group
+            //println "Parameters:\n" //nested access to the Parameters
+            def sub_params = new LinkedHashMap()
+            submap.each { innerkey, value ->
+                //println "$innerkey\t$value.description"
+                sub_params.put("$innerkey", "$value.description")
```

Comment:
Can we add the variable type here? eg. `input [str]`

Reply:
Yep. Should be able to. I havent actually touched any of the logic for reading the params in. Just beautified it for printing to screen.

Reply:
Yup, figured - feel free to ignore then 🙈